app/src/main/java/emu/skyline/adapter/AppAdapter.kt

```diff
@@ -7,80 +7,42 @@ package emu.skyline.adapter
 
 import android.app.Dialog
 import android.content.Context
-import android.graphics.Bitmap
 import android.graphics.Color
 import android.graphics.drawable.ColorDrawable
-import android.net.Uri
+import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.view.Window
 import android.widget.ImageView
+import android.widget.LinearLayout
 import android.widget.RelativeLayout
 import android.widget.TextView
+import androidx.core.content.ContextCompat
 import androidx.core.graphics.drawable.toBitmap
 import androidx.recyclerview.widget.RecyclerView
 import emu.skyline.R
-import emu.skyline.adapter.ElementType.Header
-import emu.skyline.adapter.ElementType.Item
-import emu.skyline.loader.AppEntry
-
-/**
- * This class is a wrapper around [AppEntry], it is used for passing around game metadata
- */
-class AppItem(val meta : AppEntry) : BaseItem() {
-    /**
-     * The icon of the application
-     */
-    val icon : Bitmap?
-        get() = meta.icon
-
-    /**
-     * The title of the application
-     */
-    val title : String
-        get() = meta.name
-
-    /**
-     * The string used as the sub-title, we currently use the author
-     */
-    val subTitle : String?
-        get() = meta.author
-
-    /**
-     * The URI of the application's image file
-     */
-    val uri : Uri
-        get() = meta.uri
-
-    /**
-     * The format of the application ROM as a string
-     */
-    private val type : String
-        get() = meta.format.name
-
-    /**
-     * The name and author is used as the key
-     */
-    override fun key() : String? {
-        return if (meta.author != null) meta.name + " " + meta.author else meta.name
-    }
-}
+import emu.skyline.data.AppItem
 
 /**
  * This enumerates the type of layouts the menu can be in
  */
-enum class LayoutType {
-    List,
-    Grid,
+enum class LayoutType(val layoutRes : Int) {
+    List(R.layout.app_item_linear),
+    Grid(R.layout.app_item_grid),
+    GridCompact(R.layout.app_item_grid_compact)
 }
 
+private typealias AppInteractionFunc = (appItem : AppItem) -> Unit
```

Comment:
`InteractionFunction` is better here, we tend to not use short-forms in type names unless we're being faithful to an external reference document.